### PR TITLE
Optimize: remove some unneeded checks when moving tablet to trash

### PR DESCRIPTION
**Pull request by sduzh**
1. There is no need for `FileUtils::create_dir()` to check if the parent directory exists.
2. There is no need to check if a directory is empty before removing it.
3. There is no need to call `FileUtils::create_dir` in most cases to create the tablet trash directory.
4. There is no need to check whether a directory exists before moving it to the trash.

### CODE REVIEW

**Comment by sduzh** [comment]
run starrocks_be_unittest

File: be/src/util/file_utils.cpp [hunk 1]
```diff
@@ -51,33 +51,10 @@ Status FileUtils::create_dir(const std::string& path, Env* env) {
 
     std::string partial_path;
     for (const auto& it : p) {
-        partial_path = partial_path + it.string() + "/";
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/util/file_utils.cpp:54]: I believe so, the `Env::create_dir_if_missing()` will call stat(2) to check whether an already existing path is a directory, and stat(2) will follow the symbolic link, according to the man manual.

[the remark this answers is not in the document]
Comment (imay) [reply be/src/util/file_utils.cpp:54]: Could you test it to verify if it works correctly?

File: be/src/storage/base_tablet.h [hunk 2]
```diff
@@ -42,8 +42,12 @@ class BaseTablet : public std::enable_shared_from_this<BaseTablet> {
     virtual ~BaseTablet() = default;
 
     inline DataDir* data_dir() const;
+
+    // TODO: may be schema_hash_path() is a more accurate name?
     const std::string& tablet_path() const;
 
+    std::string tablet_id_path() const;
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/storage/base_tablet.h:63]: updated, PTAL

File: be/src/util/file_utils.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/util/file_utils.cpp:54]: I've tested it, it works fine.

**Review by imay — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_clang-format

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Review by chaoyli — APPROVED** [review]

**Pull request merged**